lib/simplecov/configuration.rb

```diff
@@ -318,16 +326,21 @@ def maximum_coverage_drop(coverage_drop = nil)
     # Default is 0% (disabled)
     #
     def minimum_coverage_by_file(coverage = nil)
-      minimum_possible_coverage_exceeded("minimum_coverage_by_file") if coverage && coverage > 100
-      @minimum_coverage_by_file ||= (coverage || 0).to_f.round(2)
+      return @minimum_coverage_by_file ||= {} unless coverage
+
+      coverage = {DEFAULT_COVERAGE_CRITERION => coverage} if coverage.is_a?(Numeric)
+
+      validate_coverage!(coverage, "minimum_coverage_by_file")
+
+      @minimum_coverage_by_file = coverage
     end
 
     #
     # Refuses any coverage drop. That is, coverage is only allowed to increase.
     # SimpleCov will return non-zero if the coverage decreases.
     #
-    def refuse_coverage_drop
-      maximum_coverage_drop 0
+    def refuse_coverage_drop(criteria = [DEFAULT_COVERAGE_CRITERION])
```

Comment:
This looks like it doesn't fit the docs you wrote, i.e. `refuse_coverage_drop :line, :branch` should fail here or I'm wrong. We need a splat here but not sure how that works with a default argument. 

I guess we'd need to do something like this: https://stackoverflow.com/a/15797441